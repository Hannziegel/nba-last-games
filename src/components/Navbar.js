import * as React from 'react';
import { NavLink, useLocation } from 'react-router-dom';
import { IoArrowBackCircle } from 'react-icons/io5';
import nbalogo from '../assets/imgs/nba-logo-2.png';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="nav container">
    <div className="nav-configurations">
      <img className="league-img" alt="League" src={nbalogo} />
    </div>
    <NavLink to="/teams" className="arrow-back">
      {useLocation().pathname === '/teams' ? 'Teams' : <IoArrowBackCircle className="icon-arrow-back" />}
    </NavLink>
  </nav>
);

export default Navbar;
