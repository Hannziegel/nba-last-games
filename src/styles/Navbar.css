.nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2vh 9vw;
}

ul {
  display: flex;
  gap: 10px;
}

li {
  list-style: none;
}

.arrow-back,
.nav-configurations {
  padding: 10px;
  cursor: pointer;
}
