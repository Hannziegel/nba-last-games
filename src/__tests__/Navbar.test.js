import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../Redux/configureStore'; // eslint-disable-line
import Navbar from '../App';

describe('App js testing', () => {
  test('Renders App', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
