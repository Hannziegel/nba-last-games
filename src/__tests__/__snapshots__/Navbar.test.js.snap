// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App js testing Renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav container"
      >
        <a
          class="arrow-back"
          href="/teams"
        >
          <svg
            class="icon-arrow-back"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 256c0 114.87 93.13 208 208 208s208-93.13 208-208S370.87 48 256 48 48 141.13 48 256zm212.65-91.36a16 16 0 01.09 22.63L208.42 240H342a16 16 0 010 32H208.42l52.32 52.73A16 16 0 11238 347.27l-79.39-80a16 16 0 010-22.54l79.39-80a16 16 0 0122.65-.09z"
            />
          </svg>
        </a>
        <ul
          class="nav-ul"
        >
          <li
            class="nav-li"
          >
            <a
              class=""
              href="/teams"
            >
              TEAMS
            </a>
          </li>
        </ul>
        <div
          class="nav-configurations"
        >
          <svg
            class="icon-search"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 64C150.13 64 64 150.13 64 256s86.13 192 192 192 192-86.13 192-192S361.87 64 256 64zm91.31 283.31a16 16 0 01-22.62 0l-42.84-42.83a88.08 88.08 0 1122.63-22.63l42.83 42.84a16 16 0 010 22.62z"
            />
            <circle
              cx="232"
              cy="232"
              r="56"
            />
          </svg>
        </div>
      </nav>
      <div
        class="page-container container"
      >
        <div
          class="league-container"
        >
          <img
            alt="League"
            class="league-img"
            src="nba-logo-2.png"
          />
          <div
            class="league-info-container"
          >
            <h2
              class="league-name"
            >
              error
            </h2>
            <p
              class="league-number-teams"
            >
              1
               Teams
            </p>
          </div>
        </div>
        <div
          class="section-title"
        >
          Teams in NBA
        </div>
        <input
          class="searchBar"
          placeholder="Search..."
          type="search"
          value=""
        />
        <div
          class="teams-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav container"
    >
      <a
        class="arrow-back"
        href="/teams"
      >
        <svg
          class="icon-arrow-back"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M48 256c0 114.87 93.13 208 208 208s208-93.13 208-208S370.87 48 256 48 48 141.13 48 256zm212.65-91.36a16 16 0 01.09 22.63L208.42 240H342a16 16 0 010 32H208.42l52.32 52.73A16 16 0 11238 347.27l-79.39-80a16 16 0 010-22.54l79.39-80a16 16 0 0122.65-.09z"
          />
        </svg>
      </a>
      <ul
        class="nav-ul"
      >
        <li
          class="nav-li"
        >
          <a
            class=""
            href="/teams"
          >
            TEAMS
          </a>
        </li>
      </ul>
      <div
        class="nav-configurations"
      >
        <svg
          class="icon-search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 64C150.13 64 64 150.13 64 256s86.13 192 192 192 192-86.13 192-192S361.87 64 256 64zm91.31 283.31a16 16 0 01-22.62 0l-42.84-42.83a88.08 88.08 0 1122.63-22.63l42.83 42.84a16 16 0 010 22.62z"
          />
          <circle
            cx="232"
            cy="232"
            r="56"
          />
        </svg>
      </div>
    </nav>
    <div
      class="page-container container"
    >
      <div
        class="league-container"
      >
        <img
          alt="League"
          class="league-img"
          src="nba-logo-2.png"
        />
        <div
          class="league-info-container"
        >
          <h2
            class="league-name"
          >
            error
          </h2>
          <p
            class="league-number-teams"
          >
            1
             Teams
          </p>
        </div>
      </div>
      <div
        class="section-title"
      >
        Teams in NBA
      </div>
      <input
        class="searchBar"
        placeholder="Search..."
        type="search"
        value=""
      />
      <div
        class="teams-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
